src/layouts/reservation-page.tsx

```diff
@@ -1,6 +1,22 @@
+import { useNavigate } from "react-router";
 import { SeatsMap } from "../components/select-seats";
+import { useEffect, useState } from "react";
 
 export const ReservationPage = () => {
+  // navigate to login page if no authentifictaion
+  const navigate = useNavigate();
+  const [isAuthenticated, setIsAuthenticated] = useState<boolean>(true);
```

Comment:
You don't really need state here, since the condition you're basing your state on and the action that acts on the state is all inside the useEffect

Reply:
same for all the other places you check for auth